example/components/ConnectWithSelect.tsx

```diff
@@ -4,32 +4,35 @@ import { GnosisSafe } from '@web3-react/gnosis-safe'
 import type { MetaMask } from '@web3-react/metamask'
 import { Network } from '@web3-react/network'
 import { WalletConnect } from '@web3-react/walletconnect'
-import { useCallback, useState } from 'react'
+import { useCallback, useEffect, useState } from 'react'
 
-import { CHAINS, getAddChainParameters, URLS } from '../chains'
+import { CHAINS, getAddChainParameters } from '../chains'
 
 function ChainSelect({
-  chainId,
+  activeChainId,
   switchChain,
-  displayDefault,
   chainIds,
 }: {
-  chainId: number
-  switchChain: (chainId: number) => void | undefined
-  displayDefault: boolean
+  activeChainId: number
+  switchChain: (chainId: number) => void
   chainIds: number[]
 }) {
   return (
     <select
-      value={chainId}
+      value={activeChainId}
       onChange={(event) => {
-        switchChain?.(Number(event.target.value))
+        switchChain(Number(event.target.value))
       }}
       disabled={switchChain === undefined}
     >
-      {displayDefault ? <option value={-1}>Default Chain</option> : null}
+      <option hidden disabled selected={activeChainId === undefined}>
+        Select chain
+      </option>
+      <option value={-1} selected={activeChainId === -1}>
+        Default
+      </option>
       {chainIds.map((chainId) => (
-        <option key={chainId} value={chainId}>
+        <option key={chainId} value={chainId} selected={chainId === activeChainId}>
           {CHAINS[chainId]?.name ?? chainId}
```

Comment:
nit: Does this need an optional operator? Seems like bad typing if it does, because chainIds is derived from CHAINS.